test/test_torch.py

```diff
@@ -3621,6 +3621,87 @@ def test_exponential(self, device, dtype):
         with self.assertRaises(RuntimeError):
             torch.empty((1,), device=device, dtype=dtype).exponential_(-0.5)
 
+    @dtypes(*torch.testing.get_all_fp_dtypes(include_bfloat16=False, include_half=False))
+    def test_cov(self, device, dtype):
+        shapes = [(1,), (10,),
+                  (1, 1), (10, 1), (50, 5),
+                  (1, 20), (5, 20)]
+        biases = [True, False]
+        rowvars = [True, False]
+        ddofs = [None, 0, 1]            
+
+        # special cases
+        t_list = []
+        t_list.append(torch.empty(0, dtype=dtype, device=device))
+        t_list.append(torch.tensor([0, -2, nan, 10.2, inf], dtype=dtype, device=device))
+        for t in t_list:
+            actual = torch.cov(t)
+            expected = torch.from_numpy(np.cov(t.cpu().numpy())).to(actual.dtype)
+            if t.size(0) == 0: 
+                self.assertEqual(actual, t)
+            else:
+                self.assertEqual(actual, expected)
+
+        # single input
+        for shape in shapes:
+            t = _make_tensor(shape, dtype, device)
```

Comment:
use make_tensor(), not _make_tensor()

Sorry that's confusing, _make_tensor() is just a legacy function in this file